napari/_qt/qt_mode_buttons.py

```diff
@@ -0,0 +1,32 @@
+from qtpy.QtWidgets import QRadioButton, QPushButton
+
+
+class QtModeRadioButton(QRadioButton):
+    def __init__(self, layer, button_name, mode, tool_tip, checked=False):
+        super().__init__()
+
+        self.mode = mode
+        self.layer = layer
+        self.setToolTip(tool_tip)
+        self.setChecked(checked)
+        self.setProperty('mode', button_name)
+        self.toggled.connect(self._set_mode)
+        self.setFixedWidth(28)
+
+    def _set_mode(self, bool):
+        with self.layer.events.mode.blocker(self._set_mode):
+            if bool:
+                self.layer.mode = self.mode
+
+
+class QtModePushButton(QPushButton):
+    def __init__(self, layer, button_name, tool_tip, slot=None):
```

Comment:
similar question about should more of these be keyword args @jni?

Reply:
maybe just `tooltip=None` with default to `button_name` if not provided, and maybe after button name `*`